File: Yandex/Practicum/Algorightms_and_Data_Structures/InterviewQuestions/B.Статус_200/solution.cpp
#include <iostream>
#include <vector>


void solution() {
    /*
    Time Complexity O(N)
    Memory Complexity O(1)
    */
    int n;
    std::scanf("%d", &n);
    std::vector<int> nums (n, 0);
    for (int i = 0; i < n; ++i) std::scanf("%d", &nums[i]);
    long long totalPairs = 0;
    std::vector<long long> hashmap (200, 0);
    for (int& num : nums) {
        int remainder = num % 200;
        totalPairs += hashmap[remainder];
        ++hashmap[remainder];
    }
    std::printf("%lld\n", totalPairs);
}


int main() {
    solution();
}

File: Yandex/Practicum/Algorightms_and_Data_Structures/InterviewQuestions/B.Статус_200/solution.py
from __future__ import annotations
import sys


def solution() -> None:
    """
    Time Complexity O(N)
    Memory Complexity O(1)
    """
    n: int = int(sys.stdin.readline().rstrip())
    nums: list[int] = list(map(int, sys.stdin.readline().rstrip().split()))
    total_pairs: int = 0
    hashmap: list[int] = [0] * 200
    for num in nums:
        remainder: int = num % 200
        total_pairs += hashmap[remainder]
        hashmap[remainder] += 1
    sys.stdout.write('{}\n'.format(total_pairs))


if __name__ == '__main__':
    solution()